src/mongoose_async_pools.erl

```diff
@@ -44,6 +50,27 @@ config_spec() ->
 pool_name(HostType, PoolId) ->
     persistent_term:get({?MODULE, HostType, PoolId}).
 
+-spec sync(mongooseim:host_type(), pool_id()) -> [pid()].
+sync(HostType, PoolId) ->
+    Pids = mongoose_async_pools:get_workers(HostType, PoolId),
+    rpc:pmap({?MODULE, call_sync}, [HostType], Pids).
+
+call_sync(Pid, HostType) ->
+    try
+        gen_server:call(Pid, sync)
+    catch C:E:S ->
+              ?LOG_ERROR(#{what => async_sync_failed,
+                           host_type => HostType,
+                           class => C, reason => E, stacktrace => S}),
+              erlang:raise(C, E, S)
+    end.
+
+-spec get_workers(mongooseim:host_type(), pool_id()) -> [pid()].
+get_workers(HostType, PoolId) ->
+    Pool = pool_name(HostType, PoolId),
+    [Sup] = [Pid || {_Name, Pid, supervisor, _Mods} <- supervisor:which_children(Pool)],
+    [Pid || {_Name, Pid, worker, _Mods} <- supervisor:which_children(Sup)].
```

Comment:
This is such an ugly ugly hack, damn, I had to https://github.com/inaka/worker_pool/pull/184. Will still take a while to get merged and pushed into a new version, so we can come back to this code later on.

Reply:
supervisors are not hacks, it is actually one of their features.

BTW, I am surprised they use such child_id()-s, that are not just simple time_checker, queue_manager name... but `time_checker_name(Name)`:

```
    TimeCheckerSpec =
        {TimeChecker,
         {wpool_time_checker, start_link, [Name, TimeChecker, OverrunHandler]},
         permanent,
         brutal_kill,
         worker,
         [wpool_time_checker]},
```

Could be

```
    TimeCheckerSpec =
        {time_checker,
         {wpool_time_checker, start_link, [Name, TimeChecker, OverrunHandler]},
         permanent,
         brutal_kill,
         worker,
         [wpool_time_checker]},
```

Reply:
The hack is that here you need to know the supervisor hierarchy, to know which ones are the workers and which ones are unrelated siblings.